src/lib/autoDetect.ts

```diff
@@ -0,0 +1,73 @@
+import { ExtensionContext, workspace, window, languages } from 'vscode';
+import { FileAssociations } from './types';
+const { extname } = require('path');
+
+export const initializeHubLAutoDetect = (context: ExtensionContext) => {
+  languages.setLanguageConfiguration('hubl', {
+    indentationRules: {
+      decreaseIndentPattern: /.*{%(.*?end).*%}*./,
+      increaseIndentPattern: /.*{%(?!.*end).*%}.*/,
+    },
+  });
+  const languageExtensions = {
+    '.html': 'html-hubl',
+    '.css': 'css-hubl',
+  };
+  const getHublPreference = context.workspaceState.get('USE_HUBL');
+
+  const filesConfig = workspace.getConfiguration('files');
+  const fileAssoc = filesConfig.get('associations') as FileAssociations;
+  context.workspaceState.update('USE_HUBL', null);
+
+  const showHublDetectedMessage = (extension: '.html' | '.css') => {
+    const hublDetectedMessage = (extension: '.html' | '.css') => {
+      return `It looks like this file contains HubL code. Would you like to use HubL for all ${extension} files in this project?`;
+    };
+    const hublDetectedYesButton = `Use HubL for ${extension} files`;
+    const hublDetectedNoButton = 'No';
+
+    window
+      .showInformationMessage(
+        hublDetectedMessage(extension),
+        hublDetectedYesButton,
+        hublDetectedNoButton
+      )
+      .then((selection) => {
+        if (selection === hublDetectedYesButton) {
+          updateWorkspaceFileAssociation(extension);
+        } else if (selection === hublDetectedNoButton) {
+          context.workspaceState.update('USE_HUBL', false);
+        }
+        handleTextDocumentOpen.dispose();
+      });
```

Comment:
OOC - what do you see happening in this case? Right now, clicking the X means that it will pop up again the next time the developer opens that workspace. I think this could be _fiiiiiiiine_ -- but I was considering a time delay on asking again. Although, I would think that if a developer was trying not to be bothered by it they would say "no" instead of using the X? Thoughts?

Reply:
I think it's fine to do nothing. Maybe could be explicit about doing nothing though. Often times I've accidentally clicked "X" on one of those pop ups and had to restart my extensions so that I could answer w/ the prompt.

Reply:
>Right now, clicking the X means that it will pop up again the next time the developer opens that workspace. 

Yep this was the behavior I got - just want to be intentional and aware that `undefined` is also a possibility here and we're handling it properly. In this case it disposes of the listener & disables the autodetection until restart, which seems fine. Other option might be to leave the listener and show it again on next detection.